rasa/core/actions/action.py

```diff
@@ -406,28 +406,26 @@ def _validate_action_result(self, result):
 
         bot_messages = []
         for response in responses:
-            if "template" in response:
-                kwargs = response.copy()
-                del kwargs["template"]
+            template = response.pop("template", None)
+            if template:
                 draft = await nlg.generate(
-                    response["template"], tracker, output_channel.name(), **kwargs
+                    template, tracker, output_channel.name(), **response
                 )
                 if not draft:
                     continue
-
-                del response["template"]
             else:
                 draft = {}
 
-            if "buttons" in response:
-                if "buttons" not in draft:
-                    draft["buttons"] = []
-                draft["buttons"].extend(response["buttons"])
-                del response["buttons"]
+            buttons = response.pop("buttons", []) or []
```

Comment:
are both `[]` necessary here?

Reply:
yes

Reply:
pop gives in an error in case the key is not there